atom/browser/ui/autofill_popup.cc

```diff
@@ -27,80 +28,97 @@ namespace atom {
 
 namespace {
 
-std::pair<int, int> CalculatePopupXAndWidth(
-    const display::Display& left_display,
-    const display::Display& right_display,
-    int popup_required_width,
-    const gfx::Rect& element_bounds,
-    bool is_rtl) {
-  int leftmost_display_x = left_display.bounds().x();
-  int rightmost_display_x =
-      right_display.GetSizeInPixel().width() + right_display.bounds().x();
-
-  // Calculate the start coordinates for the popup if it is growing right or
-  // the end position if it is growing to the left, capped to screen space.
-  int right_growth_start = std::max(
-      leftmost_display_x, std::min(rightmost_display_x, element_bounds.x()));
-  int left_growth_end =
-      std::max(leftmost_display_x,
-               std::min(rightmost_display_x, element_bounds.right()));
-
-  int right_available = rightmost_display_x - right_growth_start;
-  int left_available = left_growth_end - leftmost_display_x;
-
-  int popup_width =
-      std::min(popup_required_width, std::max(right_available, left_available));
-
-  std::pair<int, int> grow_right(right_growth_start, popup_width);
-  std::pair<int, int> grow_left(left_growth_end - popup_width, popup_width);
-
-  // Prefer to grow towards the end (right for LTR, left for RTL). But if there
-  // is not enough space available in the desired direction and more space in
-  // the other direction, reverse it.
+struct WidthCalculationResults {
```

Comment:
Did this code come from somewhere else (e.g. //chrome?)

Reply:
Yupp, that comes from https://cs.chromium.org/chromium/src/chrome/browser/ui/autofill/popup_view_common.cc?q=WidthCalculationResults&sq=package:chromium&g=0&l=31. Shall I add a comment about that?

Reply:
is there any chance we could reuse the chrome code directly rather than copying it?

Reply:
Yupp, but I had to define 2 methods that the original code uses. Is it fine if those will live in the popups's file and not under `chromium_src`?